nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -488,6 +489,74 @@ public void testForDiamondInAnAssignment() {
         .doTest();
   }
 
+  @Test
+  public void genericFunctionReturnTypeNewClassTree() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static class A<T extends @Nullable Object> { }",
+            "  static A<String> testPositive1() {",
+            "   // BUG: Diagnostic contains: Cannot return",
```

Comment:
Sure we can. I just added "Cannot return" as it is related to the function return type and we have used mismatched nullability of type parameters for the assignment as well. For assignments we are using "cannot assign" and here "cannot return" to make it more clear.

Reply:
Yup. I get that it's a little bit more likely to clash with other issues involving type parameters (which is why maybe using the full message is not a bad idea). But, if we are matching it as "Cannot return", then it can also clash with existing NullAway warnings about nullability of the return itself, without involving type parameters, and given the message is right above a return statement, that might be an easier mistake to make while reading the test case casually, rather than confusing it with a type parameter issue about something other than the value being returned...

Not super sure here, but I think "mismatched nullability of type parameters" is _slightly_ less confusing than "Cannot return" for me, even with the context of the test class name.